MoltenVK/MoltenVK/GPUObjects/MVKImage.mm

```diff
@@ -986,6 +993,35 @@
 
 	if (pExtMemInfo) { initExternalMemory(pExtMemInfo->handleTypes); }
 
+	// Setting Metal objects directly will override Vulkan settings.
+	// It is responsibility of app to ensure these are consistent. Not doing so results in undefined behavior.
+	const VkExportMetalObjectCreateInfoEXT* pExportInfo = nullptr;
+	for (const auto* next = (const VkBaseInStructure*)pCreateInfo->pNext; next; next = next->pNext) {
+		switch (next->sType) {
+			case VK_STRUCTURE_TYPE_IMPORT_METAL_TEXTURE_INFO_EXT: {
+				const auto* pMTLTexInfo = (VkImportMetalTextureInfoEXT*)next;
+				uint8_t planeIndex = MVKImage::getPlaneFromVkImageAspectFlags(pMTLTexInfo->plane);
+				setConfigurationResult(setMTLTexture(planeIndex, pMTLTexInfo->mtlTexture));
+				break;
+			}
+			case VK_STRUCTURE_TYPE_IMPORT_METAL_IO_SURFACE_INFO_EXT: {
+				const auto* pIOSurfInfo = (VkImportMetalIOSurfaceInfoEXT*)next;
+				setConfigurationResult(useIOSurface(pIOSurfInfo->ioSurface));
+				break;
+			}
+			case VK_STRUCTURE_TYPE_EXPORT_METAL_OBJECT_CREATE_INFO_EXT:
+				pExportInfo = (VkExportMetalObjectCreateInfoEXT*)next;
+				break;
+			default:
+				break;
+		}
+	}
+
+	// If we're expecting to export an IOSurface, and weren't give one,
+	// base this image on a new IOSurface that matches its configuration.
```

Comment:
```suggestion
	// If we're expecting to export an IOSurface, and weren't given one,
	// base this image on a new IOSurface that matches its configuration.
```

Reply:
We will at some point...maybe driven by the Android emulator requirements.

The use cases here were different, really about making the existing functionality, including accessing resources, but also including `MTLCommandQueue` and `MTLDevice`, Vulkan friendly.